### DISCUSSION TITLE
Deploy on AWS EBS

### DISCUSSION THREAD

**Discussion by pedromellogomes** [wheels-dev/wheels#1378] [Q&A]
I have a production app deployed on AWS elastic beanstalk. 
This application wasn't deployed by me, so I don't know much about and AWS is requiring some upgrades on that instance. 
I want to replicate the deployment on a test environment just like as if it was production but in a more updated instance.

The current production app is running on:
- Tomcat 8.5 (to be deprecated by AWS)
- Java 11
- coldfusion engine is lucee 5.3.6.61

The production app is being deployed via CodePipeline and all steps of clone, build and deploy works perfectly fine.

I started creating a new Tomcat instance and duplicate the pipeline.
But the tomcat is returning a 404 response and when i look at the logs, that's what I've found:

localhost.log
> 11-Jan-2024 19:13:00.755 SEVERE [main] org.apache.catalina.core.StandardContext.filterStart Exception starting filter [UrlRewriteFilter]
>	java.lang.NoClassDefFoundError: javax/servlet/Filter

catalina.log
> 11-Jan-2024 20:02:35.762 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server version name:   Apache Tomcat/10.1.17
11-Jan-2024 20:02:35.768 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server built:          Aug 8 2023 00:00:00 UTC
11-Jan-2024 20:02:35.769 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server version number: 10.1.17.0
11-Jan-2024 20:02:35.769 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log OS Name:               Linux
11-Jan-2024 20:02:35.769 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log OS Version:            6.1.66-91.160.amzn2023.x86_64
11-Jan-2024 20:02:35.769 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Architecture:          amd64
11-Jan-2024 20:02:35.769 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Java Home:             /usr/lib/jvm/java-17-amazon-corretto.x86_64
11-Jan-2024 20:02:35.769 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log JVM Version:           17.0.9+8-LTS
11-Jan-2024 20:02:35.769 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log JVM Vendor:            Amazon.com Inc.
11-Jan-2024 20:02:35.770 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_BASE:         /usr/share/tomcat10
11-Jan-2024 20:02:35.770 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_HOME:         /usr/share/tomcat10
11-Jan-2024 20:02:35.771 INFO [main] org.apache.catalina.core.AprLifecycleListener.lifecycleEvent The Apache Tomcat Native library which allows using OpenSSL was not found on the java.library.path: [/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib]
11-Jan-2024 20:02:36.248 INFO [main] org.apache.coyote.AbstractProtocol.init Initializing ProtocolHandler ["http-nio-8080"]
11-Jan-2024 20:02:36.345 INFO [main] org.apache.catalina.startup.Catalina.load Server initialization in [1094] milliseconds
11-Jan-2024 20:02:36.430 INFO [main] org.apache.catalina.core.StandardService.startInternal Starting service [Catalina]
11-Jan-2024 20:02:36.430 INFO [main] org.apache.catalina.core.StandardEngine.startInternal Starting Servlet engine: [Apache Tomcat/10.1.17]
11-Jan-2024 20:02:36.447 INFO [main] org.apache.catalina.startup.HostConfig.deployDirectory Deploying web application directory [/var/lib/tomcat10/webapps/ROOT]
11-Jan-2024 20:02:37.491 INFO [main] org.apache.jasper.servlet.TldScanner.scanJars At least one JAR was scanned for TLDs yet contained no TLDs. Enable debug logging for this logger for a complete list of JARs that were scanned but no TLDs were found in them. Skipping unneeded JARs during scanning can improve startup time and JSP compilation time.
11-Jan-2024 20:02:37.575 SEVERE [main] org.apache.catalina.core.StandardContext.startInternal One or more Filters failed to start. Full details will be found in the appropriate container log file
11-Jan-2024 20:02:37.576 SEVERE [main] org.apache.catalina.core.StandardContext.startInternal Context [] startup failed due to previous errors
11-Jan-2024 20:02:37.607 INFO [main] org.apache.catalina.startup.HostConfig.deployDirectory Deployment of web application directory [/var/lib/tomcat10/webapps/ROOT] has finished in [1,159] ms
11-Jan-2024 20:02:37.612 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["http-nio-8080"]
11-Jan-2024 20:02:37.636 INFO [main] org.apache.catalina.startup.Catalina.start Server startup in [1289] millisecond
I don't know if it's a instance configuration thing (which i looked into the live instance to copy the configuration) or some missing building config.

Does anyone have an idea how to do that or have a updated article to read about how to do cfwheels app deployments in AWS?

Thanks in advance

**Comment by neokoenig** [comment]
It's been years since I tried to do anyting on EBS (now on fargate) - so any suggestions may be out of date, but addressing the immediate error you've got there - it can't find UrlRewriteFilter, which will be this - https://tuckey.org/urlrewrite/

I'm not 100% sure, but I think @chapmandu  has apps on Beanstalk... (or at least, did a while ago - https://chapmandu.com/2015/06/09/deploying-a-lucee-application-to-an-amazon-elastic-beanstalk-environment/ )
